File: app-core/src/main/java/ru/micron/persistence/repository/RoleRepository.java
package ru.micron.persistence.repository;

import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import ru.micron.persistence.model.RoleEntity;

public interface RoleRepository extends JpaRepository<RoleEntity, UUID> {

  RoleEntity findByName(String name);
}

File: app-core/src/main/kotlin/ru/micron/persistence/repository/RoleRepository.kt
package ru.micron.persistence.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.micron.persistence.model.RoleEntity
import java.util.*

@Repository
interface RoleRepository : JpaRepository<RoleEntity, UUID> {
    fun findByName(name: String): RoleEntity
}